packages/cursorless-vscode/src/ide/vscode/VscodeCommandHistory.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  CommandComplete,
+  CommandHistory,
+  CommandHistoryItem,
+  Disposable,
+  FileSystem,
+} from "@cursorless/common";
+import * as fs from "fs/promises";
+import { mkdirSync } from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { VscodeIDE } from "./VscodeIDE";
+
+const dirName = "commandHistory";
+const filePrefix = "cursorlessCommandHistory";
+const settingSection = "cursorless";
+const settingName = "commandHistory";
+const settingFullName = `${settingSection}.${settingName}`;
+
+export class VscodeCommandHistory implements CommandHistory {
+  private readonly dirPath: string;
+  private readonly cursorlessVersion: string;
+  private disposable: Disposable;
+  private active: boolean = false;
+
+  constructor(ide: VscodeIDE, fileSystem: FileSystem) {
+    this.evaluateSetting = this.evaluateSetting.bind(this);
+    this.cursorlessVersion = ide.cursorlessVersion;
+    this.dirPath = path.join(fileSystem.cursorlessDir, dirName);
+
+    mkdirSync(this.dirPath, { recursive: true });
+
+    this.evaluateSetting();
+
+    this.disposable = vscode.workspace.onDidChangeConfiguration((event) => {
+      if (event.affectsConfiguration(settingFullName)) {
+        this.evaluateSetting();
+      }
+    });
+  }
+
+  async append(command: CommandComplete): Promise<void> {
+    if (!this.active) {
+      return;
+    }
```

Comment:
We can use this to bail out instead of having DisabledCommandHistory. Seems cleaner to have a single path.

The current code will not handle enabling command history in the existing session - because we already initialized to the DisabledCommandHistory which has no workspace config watcher

Reply:
Yes it does. Disabled command history is only used in testing. `vscode.workspace.onDidChangeConfiguration` watches for setting changes.